Framework/Algorithms/src/CorelliPowderCalibrationLoad.cpp

```diff
@@ -0,0 +1,164 @@
+
+#include <boost/filesystem.hpp>
+#include <boost/filesystem/operations.hpp>
+
+namespace Mantid {
+
+namespace Algorithms {
+using namespace Kernel;
+using namespace API;
+using namespace DataObjects;
+
+/// Config logger
+namespace {
+Logger logger("CorelliPowderCalibrationLoad");
+}
+
+DECLARE_ALGORITHM(CorelliPowderCalibrationLoad)
+
+/**
+ * @brief Initialization
+ *
+ */
+void CorelliPowderCalibrationLoad::init() {
+
+  // InputWorkspace
+  auto wsValidator = std::make_shared<InstrumentValidator>();
+  declareProperty(std::make_unique<WorkspaceProperty<MatrixWorkspace>>(
+                      "InputWorkspace", "", Direction::Input,
+                      PropertyMode::Mandatory, wsValidator),
+                  "CORELLI workspace to calibrate");
+
+  // Database directory
+  // NOTE:
+  //  The default path, /SNS/CORELLI/shared/database, has not been
+  //  setup yet, and the final decision is up to the CIS&IS of the
+  //  CORELLI
+  declareProperty(std::make_unique<FileProperty>("DatabaseDir",
+                                                 "/SNS/CORELLI/shared/database",
+                                                 FileProperty::Directory),
+                  "CORELLI calibration database directory");
+
+  // CalibrationTable
+  declareProperty(std::make_unique<WorkspaceProperty<TableWorkspace>>(
+                      "OutputWorkspace", "", Direction::Output),
+                  "Name of the output CORELLI calibration table");
+}
+
+/**
+ * @brief Validate algorithm inputs
+ *
+ * @return std::map<std::string, std::string>
+ */
+std::map<std::string, std::string>
+CorelliPowderCalibrationLoad::validateInputs() {
+  std::map<std::string, std::string> issues;
+  ws = getProperty("InputWorkspace");
+  std::string dbdir = getProperty("DatabaseDir");
+
+  // 1_check: input workspace is from CORELLI
+  if (ws->getInstrument()->getName() != "CORELLI") {
+    issues["InputWorkspace"] = "CORELLI only algorithm, aborting";
+  }
+
+  // 2_check: make sure there is a time stamp we can use in ws
+  if (!ws->run().hasProperty("start_time") &&
+      !ws->run().hasProperty("run_start")) {
+    issues["InputWorkspace"] = "InputWorkspace missing start time";
+  }
+
+  // 3_check: DB dir exsits
+  if (!boost::filesystem::exists(dbdir)) {
+    issues["DatabaseDir"] = "Given database dir does not exits";
+  }
+
+  return issues;
+}
+
+/**
+ * @brief
+ *
+ * @param ws: input workspace
+ * @return std::string : deduced filename with YYYYMMDD format
+ */
+std::string CorelliPowderCalibrationLoad::deduce_database_name(
+    API::MatrixWorkspace_sptr ws) {
+  std::string timeStamp{""};
+  if (ws->run().hasProperty("start_time")) {
+    timeStamp = ws->run().getProperty("start_time")->value();
+  } else {
+    timeStamp = ws->run().getProperty("run_start")->value();
+  }
+  // convert date to YYYYMMDD
+  std::string timeStampStr =
+      CorelliPowderCalibrationDatabase::convertTimeStamp(timeStamp);
+
+  // dedeuced file name
```

Comment:
small typo in "dedeuced"

Reply:
opps, I will correct it now :sweat_smile: